pageserver/src/layered_repository/storage_layer.rs

```diff
@@ -73,13 +73,24 @@ pub enum PageVersion {
 ///
 /// Data needed to reconstruct a page version
 ///
-/// 'page_img' is the old base image of the page to start the WAL replay with.
-/// It can be None, if the first WAL record initializes the page (will_init)
-/// 'records' contains the records to apply over the base image.
+/// This struct is used to communicate across calls to 'get_page_reconstruct_data'.
+/// Before first call to get_page_reconstruct_data, you can fill in 'page_img' if
+/// you have an older cached version of the page available. That can save work in
+/// 'get_page_reconstruct_data', as it can stop searching for page versions when
+/// all the WAL records going back to the cached image have been collected.
+///
+/// When get_page_reconstruct_data returns Complete, 'page_img' is set to an
+/// image of the page, or the oldest WAL record in 'records' is a will_init-type
+/// record that initializes the page without requiring a previous image.
+///
+/// If 'get_page_reconstruct_data' returns Continue, some 'records' may have
+/// been collected, but there are more records outside the current layer. Pass
+/// the same PageReconstructData struct in the next 'get_page_reconstruct_data'
+/// call, to collect more records.
 ///
 pub struct PageReconstructData {
     pub records: Vec<(Lsn, ZenithWalRecord)>,
-    pub page_img: Option<Bytes>,
+    pub page_img: Option<(Lsn, Bytes)>,
```

Comment:
Hmm, perhaps. In case of a will-init WAL record, it would be strange to have the first record in 'page_img' instead of the 'records' list, though.

cc'ing @MMeent, you also had some ideas on refactoring ZenithWalRecord.